wandb/sdk/interface/interface.py

```diff
@@ -169,10 +169,13 @@ def _publish_history(self, history):
         rec = self._make_record(history=history)
         self._publish(rec)
 
-    def publish_history(self, data, step=None, run=None):
+    def publish_history(self, data, step=None, run=None, publish_step=True):
         run = run or self._run
         data = data_types.history_dict_to_json(run, data, step=step)
         history = wandb_internal_pb2.HistoryRecord()
+        if publish_step:
+            history.step = step
```

Comment:
really, what is this magic?
i would have thought it was necessary to write:
```
if publish_step:
            history.step.num = step
```

Reply:
I hadn't realized my make proto had failed, so this is all wrong.